use ES6 modules for API code

camallen opened the pull request:
Fixes the issue of the app not working on safari. 

Avoid module race condition loads for modules, i.e. API wasn't defined when referencing in the module script.

Noting that the protected API routes, pending / upload, etc still do not work on safari as the basic auth protected API requests simply fail in safari only. Switching to use the manual auth code in superagent works fine though
```
get: function(path, defaultValue) {
    return superagent.get(`${API.host}${path}`)
    .auth('username', 'password')
    .then(function (response) {
      return JSON.parse(response.text);
    })
    .catch(function (error) {
      console.error(error);
      return defaultValue;  // Return a default value, don't throw an error
    });
  },
```

I can't find a reason why this is the case on my safari version, thoughts / help appreciated.

eatyourgreens commented on js/api.js:
```diff
@@ -64,6 +64,9 @@ const API = {
   },
 }
+
+export function api() {
```
Why not just export the API object here?

camallen replied:
I did not know i could export a variable, thought i had to export a function.

eatyourgreens replied:
`export const API = …` on the first line should work.

camallen replied:
marvellous and thank you :)

camallen committed:
use ES6 modules for API code

avoid module race condition loads for modules, i.e. API wasn't defined when referencing in the module script.

camallen committed:
export the API global variable

camallen committed:
cleanup console log

camallen committed:
export and declare in one line
